frontend/src/components/appbar/AppBar.tsx

```diff
@@ -0,0 +1,108 @@
+// React Framework imports
+import * as React from "react";
+import clsx from 'clsx';
+
+// Material-UI Functionality imports
+import AppBar from '@material-ui/core/AppBar';
+import IconButton from '@material-ui/core/IconButton';
+import { createStyles, makeStyles, 
+         useTheme, Theme } from '@material-ui/core/styles';
+import Toolbar from '@material-ui/core/Toolbar';
+import Typography from '@material-ui/core/Typography';
+
+// Material-UI Icon imports
+import AccountCircleIcon from '@material-ui/icons/AccountCircle';
+import MenuIcon from '@material-ui/icons/Menu';
+
+const drawerWidth = 240;
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    root: {
+      display: 'flex',
+    },
+    appBar: {
+      transition: theme.transitions.create(['margin', 'width'], {
+        duration: theme.transitions.duration.leavingScreen,
+        easing: theme.transitions.easing.sharp,
+      }),
+    },
+    appBarShift: {
+      marginLeft: drawerWidth,
+      transition: theme.transitions.create(['margin', 'width'], {
+        duration: theme.transitions.duration.enteringScreen,
+        easing: theme.transitions.easing.easeOut,
+      }),
+      width: `calc(100% - ${drawerWidth}px)`,
+    },
+    menuButton: {
+      marginRight: theme.spacing(2),
+    },
+    hide: {
+      display: 'none',
+    },
+    content: {
+      flexGrow: 1,
+      padding: theme.spacing(3),
+      transition: theme.transitions.create('margin', {
+        easing: theme.transitions.easing.sharp,
+        duration: theme.transitions.duration.leavingScreen,
+      }),
+      marginLeft: -drawerWidth,
+    },
+    contentShift: {
+      transition: theme.transitions.create('margin', {
+        easing: theme.transitions.easing.easeOut,
+        duration: theme.transitions.duration.enteringScreen,
+      }),
+      marginLeft: 0,
+    },
+    accountIcon: {
+      marginLeft: "auto"
+    }
+  }),
+);
+
+/**
+ * Creates the AppBar component to be displayed
+ * @prop {boolean=} open represents whether or not Drawer comp is displayed
+ * @prop {handleDrawerOpen(): void} handleDrawerOpen sets open to true
+ * @prop {handleDrawerClose(): void} handleDrawerClose sets open to false
```

Comment:
You might want to use `@param prop.something` here rather than @prop because this is a function. I like your descriptions though, they will help whoever is editing this later. (see https://jsdoc.app/tags-param.html for more info about object parameters)

Reply:
A valid point, multiple @prop statements could be misleading